### PR TITLE
Allow setting PKCS11_MODULE path via environment variable

### PR DESCRIPTION
**Pull request by klemensn**
If I read the code right, libdigidocpp and/or qdigidoc4 take
`PKCS11_MODULE` as file name and attempt to load it based on Qt's
`applicationDirPath()`.

Either way, on platforms such as OpenBSD, the current code fails to
find a suitable module and thus `digidoc-tool` and `qdigidoc4` are
unable to interact with smart cards.

Copying what `modules/VersionInfo.cmake` does, allow passing an
environment variable `PKCS11_MODULE_PATH` such that package maintainers
(like me) may pass an absoloute path to the required PKCS11 module at
build-time.

This seems suitable as environment variables can be set easily and do
not need further handling in CMakeLists.txt, as would be the case with
a proper cmake variable.

### CODE REVIEW

File: modules/FindPKCS11.cmake [hunk 1]
```diff
@@ -8,6 +8,8 @@ if(APPLE)
 	find_library(PKCS11_MODULE NAMES opensc-pkcs11.so HINTS /Library/OpenSC/lib)
 elseif(WIN32)
 	set(PKCS11_MODULE opensc-pkcs11.dll)
+elseif( $ENV{PKCS11_MODULE_PATH} )
+	set( PKCS11_MODULE $ENV{PKCS11_MODULE_PATH} )
```
Comment (metsma) [inline modules/FindPKCS11.cmake]: Can you use same code format as rest of file

Comment (klemensn) [reply modules/FindPKCS11.cmake]: Sure, I've removed the spaces if that's what you meant.

**Review by metsma — APPROVED** [review]
Thank you

**Comment by flokli** [comment]
Shouldn't it be easier/more standard to just make the `set(PKCS11_MODULE opensc-pkcs11.{so,dll})` calls conditional, and skip if `PKCS11_MODULE` cmake argument is already set?

That way, `cmake` can just be executed with `-DPKCS11_MODULE=/path/to/module.so`, without the need to set nonstandard environment variables.

**Comment by flokli** [comment]
PR at https://github.com/open-eid/cmake/pull/35.

**Pull request closed without merging**